src/PSRule.Rules.Azure/PSRule.Rules.Azure.psm1

```diff
@@ -910,6 +910,24 @@ function VisitAKSCluster {
     }
 }
 
+function VisitPublicIP {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $True, ValueFromPipeline = $True)]
+        [PSObject]$Resource,
+
+        [Parameter(Mandatory = $True)]
+        [Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer]$Context
+    )
+    process {
+        # Get-AzResource does not return zones, even with latest API version
+        # Had to fetch the zones using Get-AzPublicIpAddress and insert them into the resource
+        # Logged an issue with Az PowerShell: https://github.com/Azure/azure-powershell/issues/15905
+        $publicIpAddressZones = (Get-AzPublicIpAddress -Name $Resource.Name -ResourceGroupName $Resource.ResourceGroupName -DefaultProfile $Context).Zones;
```

Comment:
@ArmaanMcleod Nice find. Not ideal but we'll have to work around it. We should also take a dependency on this module in `PSRule.Rules.Azure.psd1`.

Reply:
@BernieWhite Added to module dependencies 🙂